hedera-node/src/main/java/com/hedera/services/legacy/services/state/AwareProcessLogic.java

```diff
@@ -85,30 +84,34 @@ public AwareProcessLogic(ServicesContext ctx) {
 	}
 
 	@Override
-	public void incorporateConsensusTxn(Transaction platformTxn, Instant consensusTime, long submittingMember) {
-		try {
-			final var accessor = new PlatformTxnAccessor(platformTxn);
-			Instant effectiveConsensusTime = consensusTime;
-			if (accessor.canTriggerTxn()) {
-				effectiveConsensusTime = consensusTime.minusNanos(1);
-			}
+	public void incorporateConsensusTxn(
+			Transaction platformTxn,
+			PlatformTxnAccessor accessor,
+			Instant consensusTime,
+			long submittingMember
+	) throws NullPointerException {
+		if (accessor == null) {
+			throw new NullPointerException("The PlatformTxnAccessor is not defined.");
+		}
 
-			if (!ctx.invariants().holdFor(accessor, effectiveConsensusTime, submittingMember)) {
-				return;
-			}
+		Instant effectiveConsensusTime = consensusTime;
+		if (accessor.canTriggerTxn()) {
+			effectiveConsensusTime = consensusTime.minusNanos(1);
+		}
 
-			ctx.expiries().purge(effectiveConsensusTime.getEpochSecond());
+		if (!ctx.invariants().holdFor(accessor, effectiveConsensusTime, submittingMember)) {
+			return;
```

Comment:
Seems like something unusual happened here, should we have a log message here before return?